### DISCUSSION TITLE
Getting joint limits

### DISCUSSION THREAD

**Discussion by matusko89** [robotology/community#9] [Q&A] [answered]
So after my  original post on robotcub 

"Hi hackers, 
I was trying to find if there is a way to get the limit for some iCub simulator joint. Something that would return the lower or upper limit of desired joint. Is there a way to do this or I need to remember this limits for each joint?"

and reply from @pattacini I am opening this issue :)

**Comment by barbalberto** [comment, marked as answer]
To get the joint limits you have to open the [IControlLimits interface](http://wiki.icub.org/yarpdoc/classyarp_1_1dev_1_1IControlLimits.html) using the `view` method.

``` cpp
yarp::dev::IControlLimits *iLim;
device.view(iLim);     // device is your remoteControlBoard
```

and then call the `getLimits` method:

``` cpp
iLim->getLimits(int j, double *min, double *max);
```

This will return the `min` and `max` bounds for the given joint `j`.

**Comment by pattacini** [comment]
To open a motor device to then use the `yarp::dev::IControlLimits` interface, refer to the [wiki](http://wiki.icub.org/iCub/main/dox/html/icub_motor_control_tutorial.html).
